mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp

```diff
@@ -1492,6 +1492,43 @@ class VectorShufflePattern
     return success();
   }
 };
+
+//===----------------------------------------------------------------------===//
+// ControlBarrierOp conversion
+//===----------------------------------------------------------------------===//
+
+class ControlBarrierPattern
+    : public SPIRVToLLVMConversion<spirv::ControlBarrierOp> {
+public:
+  using SPIRVToLLVMConversion<spirv::ControlBarrierOp>::SPIRVToLLVMConversion;
+  LogicalResult
+  matchAndRewrite(spirv::ControlBarrierOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    auto i32Ty = IntegerType::get(rewriter.getContext(), 32);
```

Comment:
auto - > Type

Reply:
Agree with Ettore that this is a temporary solution.  We shall start moving Triton MLIR GPU -> MLIR SPIRV to Triton MLIR GPU -> GENX MLIR so we don't do unnecessary work.  We can define a genx.barrier in GENX and use same SPIRV device barrier function to lower it to LLVM.

Reply:
Agree this is temporary solution, added to PR description.